src/psyclone/tests/test_files/dynamo0p3/16.4.1_multiple_scalar_sums2.f90

```diff
@@ -11,6 +12,6 @@ program single_invoke
   integer(r_def)   :: rsum1, rsum2
   type(field_type) :: f1
 
-  call invoke( inner_product(rsum1, rsum2, f1) )
+  call invoke( X_innerproduct_Y(f1, rsum1, rsum2) )
```

Comment:
Is this file still required? I think we already have one that contains multiple reductions.
I also realise that this file is named incorrectly as it only has a single scalar sum/reduction (my fault).

Reply:
I am not really sure. This test is linked with multi_reduction_builtins_mod.f90 in psyGen_test.py and it seems they have intentionally broken metadata or invokes or both (i.e. passing scalars in place of fields and vice versa). For now I corrected definitions (`real(r_def)` from the incorrect `integer(r_def)`), updated BSD license and argument order to reflect changes in re-ordering X_innerproduct_Y (write argument first).
If I remove this file then multi_reduction_builtins_mod.f90 and `test_call_multi_reduction_error(monkeypatch):` which calls them in psyGen_test.py are also obsolete. Did Rupert write this test or you? 16.4.1 file has his name.